contracts/extensions/StakedExpenditure.sol

```diff
@@ -0,0 +1,299 @@
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  The Colony Network is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with The Colony Network. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+pragma solidity 0.7.3;
+pragma experimental ABIEncoderV2;
+
+import "./../colony/ColonyDataTypes.sol";
+import "./../colonyNetwork/IColonyNetwork.sol";
+import "./../patriciaTree/PatriciaTreeProofs.sol";
+import "./ColonyExtensionMeta.sol";
+
+// ignore-file-swc-108
+
+
+contract StakedExpenditure is ColonyExtensionMeta, PatriciaTreeProofs {
+
+  // Events
+
+  event ExpenditureMadeViaStake(address indexed creator, uint256 expenditureId, uint256 stake);
+  event ExpenditureCancelled(uint256 expenditureId);
+  event StakeReclaimed(uint256 expenditureId);
+
+  // Datatypes
+
+  struct Stake {
+    address creator;
+    uint256 amount;
+  }
+
+  // Storage
+
+  uint256 stakeFraction;
+
+  mapping (uint256 => Stake) stakes;
+
+  // Modifiers
+
+  modifier onlyRoot() {
+    require(colony.hasUserRole(msgSender(), 1, ColonyDataTypes.ColonyRole.Root), "staked-expenditure-caller-not-root");
+    _;
+  }
+
+  // Overrides
+
+  /// @notice Returns the identifier of the extension
+  function identifier() public override pure returns (bytes32) {
+    return keccak256("StakedExpenditure");
+  }
+
+  /// @notice Returns the version of the extension
+  function version() public override pure returns (uint256) {
+    return 1;
+  }
+
+  /// @notice Configures the extension
+  /// @param _colony The colony in which the extension holds permissions
+  function install(address _colony) public override auth {
+    require(address(colony) == address(0x0), "extension-already-installed");
+
+    colony = IColony(_colony);
+  }
+
+  /// @notice Called when upgrading the extension
+  function finishUpgrade() public override auth {}
+
+  /// @notice Called when deprecating (or undeprecating) the extension
+  function deprecate(bool _deprecated) public override auth {
+    deprecated = _deprecated;
+  }
+
+  /// @notice Called when uninstalling the extension
+  function uninstall() public override auth {
+    selfdestruct(address(uint160(address(colony))));
+  }
+
+  // Public
+
+  function setStakeFraction(uint256 _stakeFraction) public onlyRoot {
+    require(_stakeFraction <= WAD, "staked-expenditure-value-too-large");
+    stakeFraction = _stakeFraction;
+  }
+
+  function makeExpenditureWithStake(
+    uint256 _permissionDomainId,
+    uint256 _childSkillIndex,
+    uint256 _domainId,
+    bytes memory _key,
+    bytes memory _value,
+    uint256 _branchMask,
+    bytes32[] memory _siblings
+  )
+    public
+  {
```

Comment:
On the flip side, we could check to see if the `creator` address is non-zero, which would work regardless of the stake. The choice of checking the stake was made when we had the creator as a mapping key, rather than as a value in the struct.